hooks/useChainCurrencies.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  AddressFilter,
+  CurrencyFilter,
+  IntFilter,
+  useChainCurrenciesQuery,
+} from '../graphql'
+
+export default function useChainCurrencies(
+  chainId?: number,
+  { onlyERC20 }: { onlyERC20?: boolean } = {},
+): ReturnType<typeof useChainCurrenciesQuery> {
+  const filter = {
+    chainId: { equalTo: chainId } as IntFilter,
+  } as CurrencyFilter
+
+  if (onlyERC20) filter['address'] = { isNull: false } as AddressFilter
+
+  return useChainCurrenciesQuery({
+    variables: { filter },
+    skip: chainId === undefined,
+  })
```

Comment:
Could simplify returned data structure as this is a custom hook and not autogenerated Apollo hook:
```suggestion
  const {data} = useChainCurrenciesQuery({
    variables: { filter },
    skip: chainId === undefined,
  })
  return data?.currencies?.nodes
```
This will avoid to have this kind of useMemo:
```ts
  const currencies = useMemo(
    () => chainCurrency.data?.currencies?.nodes || [],
    [chainCurrency],
  )
```